AG-14318 Convert change-detection lint script to ESLint Rule

olegat opened the pull request:
https://ag-grid.atlassian.net/browse/AG-14318

olegat committed:
AG-14318 lint-change-detection converted into an ESLintRule

olegat committed:
AG-14318 Fix change-detection.mjs

olegat committed:
AG-14318 Enable aglint/change-detection by default

olegat committed:
AG-14318 Add new lint-change-detection.test.ts

olegat committed:
AG-14318 Add a snapshot-based lint-change-detection.test.ts implementation

olegat committed:
AG-14318 Fix lint-change-detection.test.ts config & add snapshots

olegat committed:
AG-14318 Remove old change-detection tool

olegat committed:
AG-14318 Add lint-change-detection test files to global ignores

olegat committed:
AG-14318 Ignore test *.mjs files in ag-charts-community:lint:depcruise

olegat committed:
AG-14318 Fix change-detection.mjs on optional tuples

olegat committed:
AG-14318 Replace __dirname with constant string

olegat committed:
AG-14318 nx format

alantreadway reviewed (commented):
I'd strongly prefer it if we didn't have tooling-related code in `ag-charts-community`.

olegat committed:
Revert "AG-14318 Ignore test *.mjs files in ag-charts-community:lint:depcruise"

This reverts commit 1ebc0400f9d4505efdac4c20cdfe2e48be67d17d.

olegat committed:
Partially revert "AG-14318 Add a snapshot-based lint-change-detection.test.ts implementation"

This partially reverts commit 75dafe37ca7297c0d9ada555be7a69e064320fdc

olegat committed:
AG-14318 Move aglint/change-detection tests into libraries/

olegat committed:
AG-14318 Use class stubs in lint-change-detection.data.ts

olegat committed:
AG-14318 Move tsconfig.change-detection.json

olegat committed:
AG-14318 nx format

olegat committed:
AG-14318 Add project files for ag-charts-eslint-rules-test

Tested with `nx run ag-charts-eslint-rules-test:lint`

olegat committed:
AG-14318 Add ag-charts-eslint-rules-test to "all"

olegat committed:
AG-14318 Fix ag-charts-eslint-rules-test:test

olegat committed:
AG-14318 Rename ag-charts-eslint-rules-test to ag-charts-eslint-rules

olegat committed:
AG-14318 nx format

olegat committed:
AG-14318 Move eslint-rules/ into libraries/

olegat committed:
AG-14318 Remove redundant eslint config

This rule is already included in all *.ts files by the ./eslint.config.mjs root
config file.

alantreadway reviewed (approved):
LGTM - thanks for making those updates.
